### DISCUSSION TITLE
/hyndsight/forecasting-within-limits/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#168] [Announcements]
Forecasting within limits 
 https://robjhyndman.com/hyndsight/forecasting-within-limits/

**Comment by robjhyndman** [comment]
```header
Originally posted by hk on 2014-02-23 09:58:50
````
Thanks, this is really useful.

**Comment by robjhyndman** [comment]
```header
Originally posted by Sergio Ferragut on 2018-01-11 19:12:50
````
Thanks for everything you\'ve put into R.\
In the forecast constrained to an interval example, the log transform
results in -Inf values if the lower limit is zero and the data series
also contains zeros. This in turn causes the ets call to fail.

I first thought I could simply add 1 to the original series before log
transform and then subtract 1 from the result of reverse transformed
forecast. But this resulted in negative values when the reverse
transformed forecast is less than 1. So, so you have any other
suggestions on how to deal with sales forecasts where you want to avoid
negative values?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2018-01-15 22:54:40
````
If your data contain zeros, then they do not have a continuous sample
space and transformations will not easily work. You probably need a
count time series model.

**Comment by robjhyndman** [comment]
```header
Originally posted by Ravi Shankar on 2019-01-14 16:49:58
````
This is amazing. thanks a lot.\
How can we do this in python?

**Comment by csprock** [comment]
When doing model selection, it is best to compute the metrics on the transformed series, or backtransform the forecasts first then compute the metrics in the original scale?

**Reply by robjhyndman** [reply]
Usually we are interested in how accurate our forecasts are compared to the actual values, so we work on the original scale.
